yt/yt/library/auth_server/cypress_cookie_login.cpp

```diff
@@ -78,8 +78,20 @@ class TCypressCookieLoginHandler
         const IRequestPtr& req,
         const IResponseWriterPtr& rsp)
     {
-        auto replyAndLogError = [&] (const TError& error, const std::optional<std::string>& user = {}) {
-            ReplyError(rsp, error);
+        auto replyAndLogError = [&] (
+            const TError& error,
+            const std::optional<std::string>& user = {},
+            bool replyWithGenericError = false)
+        {
+            if (replyWithGenericError) {
+                // Hide details about unsuccessful login attempts for security reasons.
+                auto genericError = error;
+                genericError.SetMessage("Incorrect login or password");
```

Comment:
This can be dangerous. In YTsaurus, errors are hierarchical, i.e. error is tuple of error message, error code, some attributes and also nested errors, so errors form a tree.

Here you change the message of the root error, potentially leaving some sensitive information in the other nodes. Let's make it simple: `TError("Incorrect login or password")` will create a new error that definitely does not have anything inside.